oauth2_provider/management/commands/create_oauth2_client.py

```diff
@@ -73,13 +73,37 @@ def handle(self, *args, **options):
         self._clean_args(args)
         self._parse_options(options)
 
+        client_id = self.fields.get('client_id')
         trusted = self.fields.pop('trusted')
-        client = Client.objects.create(**self.fields)
-
+        trusted_client_exists = None
+
+        # Check if client ID is already in use. If so, fetch existing Client and update fields.
+        client_id_claimed = Client.objects.filter(client_id=client_id).exists()
```

Comment:
Updated. We can't `get_or_create` a Client using only a `client_id`, so we do need some of the complexity of what I've currently got. However, you're right that this can be simplified. While writing this, I was under the impression that using a modified Client instance to `get_or_create` a TrustedClient would create a new TrustedClient, leaving behind any old TrustedClient entry associated with the original, unmodified Client. That's not true, though.